legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/protocols/pure/vX_X_X/transfers/toPureGraph/connectionRelational.pure

```diff
@@ -0,0 +1,176 @@
+{
+  println($connectionJSON);
+
+  //TODO - proper json handling
+  let protocolConn = meta::json::fromJSON($connectionJSON, 
+                  meta::protocols::pure::vX_X_X::metamodel::store::relational::connection::RelationalDatabaseConnection ,
+                 ^JSONDeserializationConfig( 
+                                            typeKeyName='_type',
+                                             failOnUnknownProperties=false ,
+                                             typeLookup = [
+                                                            pair('delegatedKerberos', 'DelegatedKerberosAuthenticationStrategy'),
+                                                            pair('userNamePassword', 'UserNamePasswordAuthenticationStrategy'),
+                                                            pair('h2Default', 'DefaultH2AuthenticationStrategy'),
+                                                            pair('test', 'TestDatabaseAuthenticationStrategy'),
+                                                            pair('snowflakePublic', 'SnowflakePublicAuthenticationStrategy'),
+                                                            pair('gcpApplicationDefaultCredentials', 'GCPApplicationDefaultCredentialsAuthenticationStrategy'),
+
+                                                            pair('static', 'StaticDatasourceSpecification'),
+                                                            pair('h2Embedded', 'EmbeddedH2DatasourceSpecification'),
+                                                            pair('h2Local', 'LocalH2DatasourceSpecification('),
+                                                            pair('snowflake', 'SnowflakeDatasourceSpecification'),
+                                                            pair('bigQuery', 'BigQueryDatasourceSpecification')
+                                                          ]
+                                            )
+                );
+
+
+  println('\nProtocol connection parsed from json');
+  println($protocolConn);
+
+  //TODO - proper DESERIALIZATION of connection
+  let pureConn = $protocolConn->meta::protocols::pure::vX_X_X::transformation::toPureGraph::connection::transformDatabaseConnection($extensions);
+  println('\nPure connection deserialized from protocol');
+  println($pureConn);
+  $pureConn->cast(@meta::pure::alloy::connections::RelationalDatabaseConnection); 
+}
+
+
+function meta::protocols::pure::vX_X_X::transformation::toPureGraph::connection::transformDatabaseConnection(conn : meta::protocols::pure::vX_X_X::metamodel::store::relational::connection::DatabaseConnection[1] , extensions:meta::pure::router::extension::RouterExtension[*]):meta::relational::runtime::DatabaseConnection[1]
+{
+  let element = if($conn.element=='',                               // rather than root element , here it means dumy store
```

Comment:
When is element empty ?

Reply:
Legend Engine Test server sends a connection for a db type, but it doesnt know element/store beforehand. Also Jackson doesnt not allow null field , so sending '' as a workaround.